Tests/tests/test_CompositeInput.cpp

```diff
@@ -0,0 +1,339 @@
+                    return IARM_RESULT_SUCCESS;
+                }));
+
+        EXPECT_EQ(string(""), plugin->Initialize(nullptr));
+    }
+    virtual ~CompositeInputInitializedTest() override
+    {
+        plugin->Deinitialize(nullptr);
+
+        IarmBus::getInstance().impl = nullptr;
+    }
+};
+
+
+class CompositeInputInitializedEventTest : public CompositeInputInitializedTest {
+protected:
+    ServiceMock service;
+    Core::JSONRPC::Message message;
+    FactoriesImplementation factoriesImplementation;
+    PluginHost::IDispatcher* dispatcher;
+
+    CompositeInputInitializedEventTest()
+        : CompositeInputInitializedTest()
+    {
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+
+        dispatcher = static_cast<PluginHost::IDispatcher*>(
+            plugin->QueryInterface(PluginHost::IDispatcher::ID));
+        dispatcher->Activate(&service);
+    }
+
+    virtual ~CompositeInputInitializedEventTest() override
+    {
+        dispatcher->Deactivate();
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+    }
+};
+
+class CompositeInputInitializedEventDsTest : public CompositeInputInitializedEventTest {
+protected:
+    CompositeInputImplMock compositeInputImplMock;
+
+    CompositeInputInitializedEventDsTest()
+        : CompositeInputInitializedEventTest()
+    {
+        device::CompositeInput::getInstance().impl = &compositeInputImplMock;
+    }
+
+    virtual ~CompositeInputInitializedEventDsTest() override
+    {
+        device::CompositeInput::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(CompositeInputTest, RegisteredMethods)
+{
+ 
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getCompositeInputDevices")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startCompositeInput")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("stopCompositeInput")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setVideoRectangle")));
+}
+
+TEST_F(CompositeInputDsTest, getCompositeInputDevices)
+{
+    ON_CALL(compositeInputImplMock, getNumberOfInputs())
+	    .WillByDefault(::testing::Return(1));
+    ON_CALL(compositeInputImplMock, isPortConnected(::testing::_))
+	    .WillByDefault(::testing::Return(true));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getCompositeInputDevices"), _T("{}"), response));
+    EXPECT_EQ(response, string("{\"devices\":[{\"id\":0,\"locator\":\"cvbsin:\\/\\/localhost\\/deviceid\\/0\",\"connected\":\"true\"}],\"success\":true}"));
+}
+
+TEST_F(CompositeInputDsTest, startCompositeInputInvalid)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("startCompositeInput"), _T("{\"portId\": \"b\"}"), response));
+    EXPECT_EQ(response, string(""));
+}
+
+TEST_F(CompositeInputDsTest, startCompositeInput)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("startCompositeInput"), _T("{\"portId\": \"0\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+}
```

Comment:
test should verify the behavior
what is the behavior of startCompositeInput?
please describe what this api does and add expectations for that behavior.

Reply:
added selectPort() in  expectation of startCompositeInput api